### PR TITLE
Add GitHub helper functions

### PR DESCRIPTION
**Pull request by alysbrooks**
This adds a GitHub namespace with functions for retrieving LIOSS repositories. We have quite a few, so I also created two that get a subset, `get-top-lioss-repos` and `get-recent-lioss-repos`.

### CODE REVIEW

**Comment by plexus** [comment]
This is excellent! Unfortantely we do have all kinds of stuff under gh.com/lambdaisland, lots of repos that accompany old episodes for instance. Maybe we should standardize on a way to recognize which of these are actual open source projects, for instance checking for a `bin/proj` file, or putting something in the description (like a specific emoji).

**Comment by alysbrooks** [comment]
Yeah, I tried to work around this by looking for ones we highlighted in the README and the ones that were modified lately, but neither is completely accurate.

I suppose we could also use YAML front matter, but GitHub [renders those in a table](https://github.blog/2013-09-27-viewing-yaml-metadata-in-your-documents/), which is kind of obtrusive. 

Maybe we could check for the Clojars badge? "img.shields.io/clojars' I tried that out and it returned the following repositories:


```clojure 

["uniontypes" "uri" "ring.middleware.logger" "kaocha" "ansi" "kaocha-midje"
 "deep-diff2" "kaocha-junit-xml" "kaocha-cucumber" "trikl"
 "kaocha-cloverage" "zipper-viz" "open-source" "kaocha-cljs"
 "kaocha-boot" "metabase-datomic" "glogi" "logback-clojure-filter"
 "regal" "fetch" "edn-lines" "janus" "chui" "specmonstah-malli"
 "new-project" "kaocha-cljs2" "funnel" "funnel-client"
 "kaocha-cljs2-demo" "cljbox2d" "embedkit" "data-printers"
 "daedalus" "dom-types" "puck" "deja-fu"]

```

**Comment by alysbrooks** [comment]
@plexus Do you think this is ready to merge?

File: src/lioss/github.clj [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+(defn clone-repositories [repositories dir]
+  (doseq
+    [url  (map #(get % "clone_url") repositories)]
+    (try
+      (git/clone-with-cwd! url dir )
+      (catch Exception _e
+        (println _e)))))
+
+(clone-repositories (get-clojars-lioss-repositories) "..")
```
Comment (plexus) [inline src/lioss/github.clj]: I would suggest not calling this at the top level, we might want to load this namespace elsewhere for some of its functions. We could either use Clojure CLI `:exec-fn` or babashka tasks, or add `bin/proj` / `bin/dev` style wrapper like we have in other projects.

Comment (plexus) [reply src/lioss/github.clj]: Or comment it out for now, you can simply call it from a REPL.

**Comment by plexus** [comment]
I left a small comment, but generally it looks good. Also feel free to just push/merge to this repo. I think we can take shared ownership so you don't need my permission.

**Pull request merged**